test/test_auth_api.js

```diff
@@ -0,0 +1,228 @@
+import btoa       from 'btoa';
+import jwt        from 'jsonwebtoken';
+import nock       from 'nock';
+import should     from 'should';
+import supertest  from 'supertest-as-promised';
+import url        from 'url';
+
+import app        from '../src/server';
+import db         from '../src/models/db';
+import config     from '../src/config';
+import {
+  errnos,
+  ERRNO_UNAUTHORIZED,
+  errors,
+  UNAUTHORIZED
+} from '../src/errors';
+
+import {
+  loginAsAdmin,
+  loginAsClient,
+  createClient,
+} from './common';
+
+const endpointPrefix = '/' + config.get('version');
+const server = supertest(app);
+
+describe('Authentication API', () => {
+  describe('POST ' + endpointPrefix + '/auth/basic', () => {
+    it('should respond 401 Unauthorized if there is no auth header', done => {
+      server.post(endpointPrefix + '/auth/basic')
+            .expect(401)
+            .end((err, res) => {
+              res.status.should.be.equal(401);
+              res.body.code.should.be.equal(401);
+              res.body.errno.should.be.equal(errnos[ERRNO_UNAUTHORIZED]);
+              res.body.error.should.be.equal(errors[UNAUTHORIZED]);
+              done();
+            });
+    });
+
+    it('should respond 401 Unauthorized if auth header is invalid', done => {
+      server.post(endpointPrefix + '/auth/basic')
+            .set('Authorization', 'Invalid')
+            .expect(401)
+            .end((err, res) => {
+              res.status.should.be.equal(401);
+              res.body.code.should.be.equal(401);
+              res.body.errno.should.be.equal(errnos[ERRNO_UNAUTHORIZED]);
+              res.body.error.should.be.equal(errors[UNAUTHORIZED]);
+              done();
+            });
+    });
+
+    it('should respond 401 Unauthorized if admin pass is incorrect', done => {
+      server.post(endpointPrefix + '/auth/basic')
+            .set('Authorization', 'Basic invalidpassword')
+            .expect(401)
+            .end((err, res) => {
+              res.status.should.be.equal(401);
+              res.body.code.should.be.equal(401);
+              res.body.errno.should.be.equal(errnos[ERRNO_UNAUTHORIZED]);
+              res.body.error.should.be.equal(errors[UNAUTHORIZED]);
+              done();
+            });
+    });
+
+    it('should respond 201 Created if admin pass is correct', done => {
+      const pass = btoa('admin:' + config.get('adminPass'));
+      server.post(endpointPrefix + '/auth/basic')
+            .set('Authorization', 'Basic ' + pass)
+            .expect(200)
+            .end((err, res) => {
+              res.status.should.be.equal(201);
+              should.exist(res.body.token);
+              done();
+            });
+    });
+  });
```

Comment:
I only grouped them inside the same `describe` + the endpoint, no other change. There are few things we could do though: template strings, returning a promise instead of using done, etc.